src/amo/components/PermissionsCard/permissions.js

```diff
@@ -84,9 +91,12 @@ export class PermissionUtils {
         platformFiles,
       );
 
-      return [];
+      return permissions;
     }
-    return file.permissions || [];
+
+    permissions.optional = file.optional_permissions;
```

Comment:
So why are `required` permissions getting a fallback value `[]` but not `optional`?

Reply:
In our types, `permissions` is listed as `permissions?: Array<string>`, so it's apparently possible that it will be `undefined`, so we need a fallback. `optional_permissions` will always be available, so it doesn't need a fallback. I just checked the API docs, and they don't say anything about `permissions` being possibly undefined, so maybe that was a mistake way back when it was created (which was 3 years ago)? I will check the server code to see if `permissions` is actually optional, and if not, we can change the code to always expect a value for it.

Reply:
I just checked the server code, and it looks like permissions will always be at least an empty array. I have no idea why it was defined as `permissions?: Array<string>` to begin with. It looks like the code to expose `permissions` was added 4 years ago and was never able to return `undefined`, so I'll update this and the related code.